src/components/calcite-picker/calcite-picker-row/calcite-picker-row.tsx

```diff
@@ -89,7 +94,7 @@ export class CalcitePickerRow {
     if (!this.icon) {
       return null;
     }
-    if (this.icon === "grip") {
+    if (this.icon === ICON_TYPES["grip"]) {
```

Comment:
Should we be using a type here instead of a constant?

```ts
type PickerIcon = "square" | "circle" | "grip";

@Prop({ reflect: true }) icon: PickerIcon:
```

That way if something other than one of those is set to the icon it will display an error.

I think the current way it is it won't be able to determine the value since its not using dot noation `ICON_TYPES.square`

You can still have the constant for use within the tests but for defining the type, I think it should be using a string???

My only concern is to make sure the typings are getting picked up correctly.

Reply:
A string enum seems to just compile down to a pojo. Would that help with typing here in some way I'm not aware of?
http://www.typescriptlang.org/play/#code/KYOwrgtgBAIglgJ2AYwC5wPYigbwFBSFQCqADlALxQBExACtQDQFEwYDu2V1MA8gOoA5Ji0IAZYADNUlGmICiAMQAqIolABKcAOYALGdw0BJAOIAJVcwC+QA

@driskull I can use dot notation if that helps the typescript compiler. Can you guys help me understand what typing error you guys are trying to avoid? Users aren't able to set the icon, it's all controlled internally from the picker.

Reply:
Going with enum - works fine.